modules/javafx.graphics/.classpath

```diff
@@ -1,6 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="src" path="build/gensrc/jsl-prism"/>
+	<classpathentry kind="src" path="build/gensrc/jsl-decora"/>
```

Comment:
As with any multi-stage project, we need to run 'gradle sdk' (or perhaps 'gradle sdk apps' for good measure) to build these directories.  Then, in Eclipse, clean all projects.  These steps produce a working configuration.

I do not see hlsl folders being build on Mac.  Are these being built on windows/linux?
If these are platform-specific, then we should add them to the gradle build.  Maybe an additional target that creates the empty directories, and upon which sdk depends on.

Reply:
As @kleopatra noted below, you need to add the hlsl folders that are built on Windows. Add them as an optional source to avoid errors on other OS's:
```
<classpathentry kind="src" path="build/hlsl/Prism">
	<attributes>
		<attribute name="ignore_optional_problems" value="true"/>
	</attributes>
</classpathentry>
```
And the same for
```<classpathentry kind="src" path="build/gensrc/jsl-prism"/>```
```<classpathentry kind="src" path="build/hlsl/Decora">```
```<classpathentry kind="src" path="build/gensrc/jsl-decora"/>```

This will allow to run dependent projects on Windows and won't cause build issues on other OS's as Eclipse ignores errors on these. There is no need to create empty folders.

Reply:
I agree that an optional dependency for a platform-specific generated source directory is a much better solution than creating an empty dir.

Reply:
I don't know what the platform-specific ones are for Mac. I can check on Linux if there's a problem running dependent projects with/without the sources.

Reply:
@nlisker : 
adding sources with <attribute name="ignore_optional_problems" value="true"/> *does not* remove the warning.
also, this warning cannot be turned off.

Reply:
@kevinrushforth  : 
Eclipse does not support conditionals in the .classpath files.  The warning cannot be turned off.  I see no other way but to create empty folders.

Reply:
On Linux there is no such warning., neither is there on Windows if I add a non-existing folder (to simulate the ones for other OS's). I posted the screenshots on the mailing list. If Eclipse for Mac is the only one producing these warnings, a bug should be filed with Eclipse JDT.

Reply:
I respectfully disagree.  
I see no harm in creating empty directories in this case if it solves the problem.  
We can log a bug against eclipse, and it will be fixed in several years, if ever.

Reply:
This is up to Kevin. It's changing the build file specifically to remove a warning for Eclipse Mac users. I don't mind. Might need to do the same for the other OS-specific folders.

Reply:
While not ideal, I suppose it's OK. I'm still amazed that Eclipse doesn't have the notion of optional source paths. How do people deal with this when debugging the JDK? Many modules (`java.base` and `java.desktop`, just to name two) have entire directories of platform-specific sources.

I will review the changes to `build.gradle`.

Reply:
It does have optional source paths, this is what I showed above:
```xml
<classpathentry kind="src" path="build/hlsl/Prism">
	<attributes>
		<attribute name="ignore_optional_problems" value="true"/>
	</attributes>
</classpathentry>
```
I never had any issue with optional sources. According to Andy, it's only on Mac that Eclipse generates a warning (though not an error?), on Win and Linux there is no issue.